### PR TITLE
Implement linearRegression natively

### PR DESCRIPTION
**Pull request by pub-djedruszczak**
Last one I swear...

**Describe your changes**
The PR implements the `linearRegression` function natively.

The code follows the [Python implementation](https://github.com/graphite-project/graphite-web/blob/a89711135b0a0a3dfa72310188e2cd33914a6674/webapp/graphite/render/functions.py#L4199-L4260).

**Testing performed**
A basic test case is defined which follows the [Python test](https://github.com/graphite-project/graphite-web/blob/master/webapp/tests/test_functions.py#L4321-L4361). Some additional cases test normalization and relative times.

**Additional context**
Add any other context about your contribution here.

### CODE REVIEW

**Comment by pub-djedruszczak** [comment]
I kept remaking this PR because I was trying to bring in the benchmark fix to avoid a merge conflict when creating a release, but I give up. It'll be way easier to just make that change to the feature branch once I open the PR to the upstream.

File: expr/func_linearregression.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+	"github.com/grafana/metrictank/api/models"
+	"github.com/grafana/metrictank/api/tz"
+	"github.com/grafana/metrictank/mdata"
+	"github.com/grafana/metrictank/schema"
+)
+
+type FuncLinearRegression struct {
+	in GraphiteFunc
+
+	startSourceAt string // at(1) format
```
Comment (shanson7) [inline expr/func_linearregression.go]: Not sure we should reference `at(1)` here. Better to say `Render Time format` probabky.

File: expr/funcs.go [hunk 2]
```diff
@@ -92,6 +92,7 @@ func init() {
 		"invert":                       {NewInvert, true},
 		"isNonNull":                    {NewIsNonNull, true},
 		"keepLastValue":                {NewKeepLastValue, true},
+		"linearRegression":             {NewLinearRegression, false},
```
Comment (shanson7) [inline expr/funcs.go]: Why `false`?

**Review by shanson7 — CHANGES_REQUESTED** [review]
Make sure to update `docs/graphite.md`

File: expr/func_linearregression.go (hunk 1 above)
Comment (pub-djedruszczak) [reply expr/func_linearregression.go]: Technically it's already referencing `at(1)` here (the `At` suffixes which come from the argument name).

File: expr/funcs.go (hunk 2 above)
Comment (pub-djedruszczak) [reply expr/funcs.go]: I figured we could leave it unstable for a while until we get a lot of confidence from real users. I'm confident the behavior is identical from my own testing with mtcmp though.

File: expr/func_linearregression.go (hunk 1 above)
Comment (pub-djedruszczak) [reply expr/func_linearregression.go]: Actually I guess it's pretty ambiguous as to whether they mean `at(1)` format or "start the source _at_ this time"

File: expr/funcs.go (hunk 2 above)
Comment (shanson7) [reply expr/funcs.go]: No one will use it if it's unstable as they would need to send the process flag and you can't do that through grafana.

**Review by shanson7 — APPROVED** [review]
89 commits. That's impressive.

**Pull request merged**